openedx/core/djangoapps/course_groups/cohorts.py

```diff
@@ -202,7 +206,8 @@ def get_cohort(user, course_key, assign=True, use_cached=False):
             assignment_type=CourseCohort.RANDOM
         ).course_user_group
 
-    user.course_groups.add(cohort)
+    membership = CohortMembership(course_user_group=cohort, user=user)
+    membership.save()
```

Comment:
This is the way all cohort memberships should be added. Previously, it varied as to whether the `user` would add a new entry to its `course_groups`, or the `course_group` would add a new entry to its `user`. Only one way to do it now.